lib/prime.rb

```diff
@@ -145,6 +145,42 @@ def prime?(value, generator = Prime::Generator23.new)
     end
   end
 
+  # Miller-Rabin Test to check if a given number is probably prime
+  #  or composite
+  # Returns true if n is probably prime, false if n is composite
+  # == Parameters
+  #
+  #  +n+:: an arbitary integer to be checked
+  #  +k+:: optional, A parameter that determines the accuracy
+  #                  of the test
+  #  References: https://en.wikipedia.org/wiki/Miller%E2%80%93Rabin_primality_test
+  #              http://rosettacode.org/wiki/Miller-Rabin_primality_test#Ruby
+  def probably_prime?(n, k = 30)
+    raise ArgumentError, 'n should be >= 3' if n < 3
```

Comment:
It's nice to keep comparable algorithms with comparable APIs.

So anyone who has currently calling `Prime.prime?(n)` should get the same data type (and presence/lack/type of exception, if appropriate) as what they would get if they switch to calling `Prime.probably_prime?(n)`, for any type and value of `n`